File: VB/Reporting_how-to-implement-a-master-detail-report-using-subreports-xrsubreport-without-t262615/Reports/MasterReport.vb
Imports System
Imports DevExpress.XtraReports.UI

Namespace dxSample
	Partial Public Class MasterReport
		Inherits DevExpress.XtraReports.UI.XtraReport

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub xrSubreport1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrSubreport1.BeforePrint
			Dim xrSubReport As XRSubreport = DirectCast(sender, XRSubreport)
			Dim subRep As DetailReport = TryCast(xrSubReport.ReportSource, DetailReport)
			subRep.Parameters("CatID").Value = Convert.ToInt32(GetCurrentColumnValue("CategoryID"))
		End Sub
	End Class
End Namespace

File: CS/Reporting_how-to-implement-a-master-detail-report-using-subreports-xrsubreport-without-t262615/Reports/MasterReport.cs
using System;
using DevExpress.XtraReports.UI;

namespace dxSample {
    public partial class MasterReport : DevExpress.XtraReports.UI.XtraReport {
        public MasterReport() {
            InitializeComponent();
        }

        private void xrSubreport1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            XRSubreport xrSubReport = (XRSubreport)sender;
            DetailReport subRep = xrSubReport.ReportSource as DetailReport;
            subRep.Parameters["CatID"].Value = Convert.ToInt32(GetCurrentColumnValue("CategoryID"));
        }
    }
}
